lib/Client/ClientTracer.php

```diff
@@ -462,7 +452,7 @@ public function _finishSpan(ClientSpan $span) {
         }
 
         $span->setEndMicros($this->_util->nowMicros());
-        $full = $this->pushWithMax($this->_spanRecords, $span, $this->_options["max_span_records"]);
+        $full = $this->_util->pushWithMax($this->_spanRecords, $span, $this->_options["max_span_records"]);
```

Comment:
I know this isn't changing any behavior, but I do wonder why we bother randomizing which one we discard. Seems easier to choose one deterministically or just drop the span that we are trying to add.

Reply:
i wanted to remove the oldest item (aka the one at the front), but this would be O(n) so I am compromising and choosing to not add the new item.